src/schema/posts.ts

```diff
@@ -1302,40 +1231,8 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       { id, reason, comment, tags }: ReportPostArgs,
       ctx: AuthContext,
     ): Promise<GQLEmptyResponse> => {
-      if (!reportReasons.has(reason)) {
-        throw new ValidationError('Reason is invalid');
-      }
-
-      if (reason === 'IRRELEVANT' && !tags?.length) {
-        throw new ValidationError('You must include the irrelevant tags!');
-      }
-
-      const added = await saveHiddenPost(ctx.con, {
-        userId: ctx.userId,
-        postId: id,
-      });
+      await reportPost({ ctx, id, reason, comment, tags });
```

Comment:
To keep backwards compatibility, we are keeping this mutation but using the new function now.